docs/lesson-plans/week-5.md

```diff
@@ -0,0 +1,639 @@
+
+  public SampleDTO(String name, Integer id) {
+    this.name = name;
+    this.id = id;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public String getName() {
+    return this.name;
+  }
+
+  // Sometimes this is included for making sure the object is valid (see below, really only done on incoming requests).
+  public void validate() {
+    if (name == null) {
+      throw ...
+    }
+    ...
+  }
+  ...
+```
+
+They're especially useful with REST requests and APIs because they can hold data from your incoming and outgoing 
+requests.
+
+### Marshalling/Unmarshalling
+
+Marshalling and Unmarshalling is the process of transforming an object instance into a serialized representation  
+of the object or vice versa. For our purposes, it's the process of turning a JSON into one of our DTOs.
+There are a couple of libraries that are really useful for this, namely [Jackson](https://github.com/FasterXML/jackson)
+and Vertx's [JsonObject](https://vertx.io/docs/apidocs/io/vertx/core/json/JsonObject.html) (what we'll be using 
+in this project). Most marshallers/unmarshallers expect the names of the JSON and the class you're working with to 
+match.
+
+The following examples use the `SampleDTO` from above.
+
+An example of Marshalling:
+```java 
+SampleDTO dto = new SampleDTO("me", 1);
+
+String json = JsonObject.mapFrom(dto).encode(); // returns "{'name':'me','id':1}"
+```
+
+An example of Unmarshalling:
+```java
+String input = "{'name':'me','id':1'}";
+
+SampleDTO dto = new JsonObject(input).mapTo(SampleDTO.class); // returns the unmarshalled object
+```
+
+>Note: In the `SampleDTO` class above, there is an empty, private constructor. This is required by your unmarshallers
+>(as far as we're aware, both JsonObject and Jackson require it) to instantiate a class before using 
+>[introspection/reflection](https://www.oracle.com/technical-resources/articles/java/javareflection.html) to find all of 
+>the required fields and load them into the class.
+
+### Validating Unmarshalled Objects
+
+After unmarshalling an object, it's important that you validate it to make sure that the fields have valid data. This
+can be just making sure that integers have a positive values, no fields are null, strings aren't empty, or whatever
+other basic data validation checks you need to perform. This needs to be done because users of your platform can't be
+trusted to enter good data. It's always a good idea to make sure the data you are provided from a public facing API is 
+what you're expecting, or you can find errors popping up in your projects. 
+
+To make sure your DTOs have safe data inside, a `validate()` method is usually created in your DTOs (and it's
+not a bad idea to put that into an interface which your DTOs will inherit). This method will be called after 
+unmarshalling (or marshalling if you're worried about returning bad data to the front end) to check the DTO's fields and
+validate them.
+
+## New Request Types
```

Comment:
I think this section should go above the DTOs since it motivates why/when we would marshal from JSON to a DTO